static/app/views/settings/components/forms/controls/rangeSlider.tsx

```diff
@@ -46,160 +48,159 @@ type Props = {
+    updateSliderValue();
+  }, [value]);
+
+  function updateSliderValue() {
+    if (!defined(value)) {
+      return;
     }
-  }
 
-  getActualValue = (sliderValue: State['sliderValue']): number | '' => {
-    const {allowedValues} = this.props;
-    let value: number | '';
+    const newSliderValueIndex = allowedValues?.indexOf(Number(value || 0)) ?? -1;
 
-    if (allowedValues) {
-      // If `allowedValues` is defined, then `sliderValue` represents index to `allowedValues`
-      value = allowedValues[sliderValue];
-    } else {
-      value = sliderValue;
+    // If `allowedValues` is defined, then `sliderValue` represents index to `allowedValues`
+    if (newSliderValueIndex > -1) {
+      setSliderValue(newSliderValueIndex);
+      return;
     }
 
-    return value;
-  };
-
-  setValue = value => {
-    this.setState({
-      sliderValue: value,
-    });
-  };
+    setSliderValue(value);
+  }
 
-  changeValue = (value, e) => {
-    if (this.props.onChange) {
-      this.props.onChange(this.getActualValue(value), e);
+  function getActualValue(newSliderValue: Props['value']): Props['value'] {
+    if (!allowedValues) {
+      return newSliderValue;
     }
-  };
 
-  handleInput = e => {
-    const sliderValue = parseInt(e.target.value, 10);
-    this.setValue(sliderValue);
-    this.changeValue(sliderValue, e);
-  };
+    // If `allowedValues` is defined, then `sliderValue` represents index to `allowedValues`
+    return allowedValues[newSliderValue];
+  }
 
-  handleBlur = e => {
-    const {onBlur} = this.props;
+  function handleInput(e: ChangeEvent<HTMLInputElement>) {
+    const newSliderValue = parseInt(e.target.value, 10);
+    setSliderValue(newSliderValue);
+    onChange?.(getActualValue(newSliderValue), e);
+  }
+
+  function handleCustomInputChange(e: ChangeEvent<HTMLInputElement>) {
+    const newSliderValue = parseInt(e.target.value, 10);
+    setSliderValue(isNaN(newSliderValue) ? 0 : value);
+  }
+
+  function handleBlur(e: MouseEvent<HTMLInputElement> | KeyboardEvent<HTMLInputElement>) {
     if (typeof onBlur !== 'function') {
       return;
     }
 
     onBlur(e);
-  };
-
-  handleCustomInputChange = e => {
-    const value = parseInt(e.target.value, 10);
-    this.setValue(isNaN(value) ? 0 : value);
-  };
-
-  handleCustomInputBlur = e => {
-    this.handleInput(e);
-  };
-
-  render() {
-    let {min, max, step} = this.props;
-    const {
-      name,
-      disabled,
-      allowedValues,
-      formatLabel,
-      placeholder,
-      showCustomInput,
-      className,
-    } = this.props;
-    const {sliderValue} = this.state;
-    let actualValue = sliderValue;
-    let displayValue: React.ReactNode = actualValue;
-
-    if (allowedValues) {
-      step = 1;
-      min = 0;
-      max = allowedValues.length - 1;
-      actualValue = allowedValues[sliderValue];
-      displayValue =
-        typeof actualValue !== 'undefined' ? actualValue : t('Invalid value');
+  }
+
+  function getSliderData() {
```

Comment:
Does this even need to be a function?

Reply:
ah, it is probably a bit cleaner

Reply:
I could write it differently, maybe with ternaries, but I think a function with returns makes it easier to read

Reply:
yeah I agree